### ISSUE TITLE
Road Disambiguation

### ISSUE THREAD

**Issue by rmidura** [EMmobility/EmDigIt#23]
Based on the newly created Road_Table, begin to disambiguate roads as differentiated by their intermediary locations. These are values found after the first and second in the Route_Comb string in the Route_Table, which is formatted as ORIGIN;DEST;INT1;INT2;INTN.

1. For every Road_ID, generate a list of intermediary locations that can be found in any matching Route_ID. Determine if any given intermediary locations exceeds 100 km from one or more other intermediary locations (as determined by their Location_Lat, Location_Long found on the Location_Table). If so, generate a new row with Road_ID + 0.1 for the corresponding Route_IDs. 

2. For every Road_ID where there are multiple Linked_Routes and their Route_ID's are sequential (ex: 3;4) create a new row equal to the last Road_ID + 0.1 (ex: 15.1). [Because these routes are presented one-after-the-other in the original texts, they are almost certainly alternative routes] 

In other words: 
IF a given Road_Location has => 2 Linked_Routes 
AND Linked_Route[2] = Linked_Route[1] + 1 (i.e. subsequent Route_ID numbering)
OR the intermediary location for Linked_Route[2] >= 100 km away from intermediary location of Linked_Route[1]
THEN create a new row with Road_ID+0.1

**Comment by isabdr** [comment]
Rachel, I'm having a hard time understanding what the intermediary locations are. Are they the other values present on the Route_Comb in the Route_Table that we were ignoring at first? If so, am I supposed to check their distance in relation to what? The origin or the final destination?
Best,
Bela

**Comment by rmidura** [comment]
Hi Bela!

Thanks for checking in. Yes, you are correct— the intermediary values are all values after the first two in the Route_Comb column. For routes that have multiple possible intermediaries (think Rome to Antwerp via Venice versus Rome to Antwerp via Milan) I would like you to measure how far apart those intermediary destinations  are from each other (Venice and Milan). If they are above the threshold, we will assume they are different roads (51.2 and 51.3 for example). 

I will try to clarify the language in the issue when I am back at my comp!

**Comment by rmidura** [comment]
Edited! Let me know if you would like to clarify in person.

**Comment by rmidura** [comment]
Hi Bela!

I hope your end of quarter is going smoothly. I'd like to make sure I have a clean copy of whatever work you've done since the last update- could you send/post a notebook before the end of the quarter? Best of luck with finals and all.

**Comment by isabdr** [comment]
Hi Rachel,

I'm so sorry it took me this long to reply, I got really caught up with all my finals, rush moving out, and flights back home. In the end, I didn't have much time to work on the last task you gave me either, so the changes I made to it are really subtle just while I was thinking how to proceed with the logic. I'm going to send you what I have anyway. The Road_Table file is the one I had finished and the Road_Disambiguation is the one I was starting to work on.
Furthermore, thanks for everything, it was a pleasure working with you during these last few weeks. I wish you all the success on your project!

Best,
Bela
________________________________

**Comment by rmidura** [comment]
Dear Bela,

Not a problem, it happens! It's been a pleasure working with you as well, and I hope you might be able to continue with us next year.

Just to check, did you mean to attach the files you mentioned? Would you mind retrying if so? 

Thanks!
RM

**Comment by isabdr** [comment]
Of course, I'm sending the files again. Please let me know if it worked!
________________________________
